File: examples/java/src/main/java/io/twitterapi/examples/CompleteExample.java
package io.twitterapi.examples;

import io.twitterapi.TwitterClient;
import io.twitterapi.config.TwitterConfig;
import io.twitterapi.model.Tweet;
import io.twitterapi.model.User;
import io.twitterapi.util.TwitterUtils;

import java.time.Instant;
import java.time.temporal.ChronoUnit;
import java.util.Map;

public class CompleteExample {
    public static void main(String[] args) {
        String apiKey = System.getenv("TWITTER_API_KEY");
        if (apiKey == null || apiKey.isEmpty()) {
            throw new IllegalArgumentException("Please set TWITTER_API_KEY environment variable");
        }

        // Initialize client with custom configuration
        TwitterConfig config = TwitterConfig.builder()
            .maxRetries(5)
            .retryInterval(Duration.ofSeconds(2))
            .build();
            
        TwitterClient client = new TwitterClient(apiKey);

        try {
            // Example 1: Get user information
            String username = "elonmusk";
            System.out.printf("%n=== Getting info for user: %s ===%n", username);
            
            User user = client.getUserInfo(username);
            System.out.printf("Name: %s%n", user.getName());
            System.out.printf("Followers: %,d%n", user.getFollowersCount());
            System.out.printf("Following: %,d%n", user.getFollowingCount());

            // Example 2: Get user's recent tweets
            System.out.println("\n=== Recent tweets ===");
            Map<String, Object> tweets = client.getUserTweets(username, false, "");
            @SuppressWarnings("unchecked")
            List<Map<String, Object>> tweetList = (List<Map<String, Object>>) tweets.get("tweets");
            
            tweetList.stream()
                .limit(3)
                .forEach(tweet -> {
                    System.out.printf("- %s...%n", 
                        ((String) tweet.get("text")).substring(0, Math.min(100, 
                            ((String) tweet.get("text")).length())));
                    System.out.printf("  Likes: %,d, Retweets: %,d%n", 
                        tweet.get("likeCount"), tweet.get("retweetCount"));
                });

            // Example 3: Advanced search
            System.out.println("\n=== Advanced Search Example ===");
            String query = TwitterUtils.buildAdvancedQuery(
                "artificial intelligence",
                "elonmusk",
                "2024-01-01",
                null,
                "en"
            );
            
            Map<String, Object> searchResults = client.searchTweets(query, "");
            System.out.printf("Search results for: %s%n", query);
            
            @SuppressWarnings("unchecked")
            List<Map<String, Object>> resultList = (List<Map<String, Object>>) searchResults.get("tweets");
            
            resultList.stream()
                .limit(3)
                .forEach(tweet -> {
                    System.out.printf("- %s...%n", 
                        ((String) tweet.get("text")).substring(0, Math.min(100, 
                            ((String) tweet.get("text")).length())));
                    System.out.printf("  Posted at: %s%n", tweet.get("createdAt"));
                });

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
            e.printStackTrace();
        }
    }
} 

File: examples/python/examples/complete_example.py
"""
Complete example demonstrating various Twitter API features.
"""

import os
import time
from datetime import datetime, timedelta
from src.client import TwitterAPIClient
from src.utils.helpers import build_advanced_query, format_timestamp

def main():
    # Initialize client
    api_key = os.getenv("TWITTER_API_KEY")
    if not api_key:
        raise ValueError("Please set TWITTER_API_KEY environment variable")
        
    client = TwitterAPIClient(api_key)
    
    # Example 1: Get user information and recent tweets
    username = "elonmusk"
    print(f"\n=== Getting info for user: {username} ===")
    
    user_info = client.get_user_info(username)
    print(f"Name: {user_info['data']['name']}")
    print(f"Followers: {user_info['data']['followers']:,}")
    print(f"Following: {user_info['data']['following']:,}")
    
    # Get recent tweets
    tweets = client.get_user_tweets(username, include_replies=False)
    print("\nRecent tweets:")
    for tweet in tweets['tweets'][:3]:  # Show first 3 tweets
        print(f"- {tweet['text'][:100]}...")
        print(f"  Likes: {tweet['likeCount']:,}, Retweets: {tweet['retweetCount']:,}")
    
    # Example 2: Advanced search
    print("\n=== Advanced Search Example ===")
    query = build_advanced_query(
        keywords="artificial intelligence",
        username="elonmusk",
        since="2024-01-01",
        lang="en"
    )
    results = client.search_tweets(query)
    print(f"\nSearch results for: {query}")
    for tweet in results['tweets'][:3]:
        print(f"- {tweet['text'][:100]}...")
        print(f"  Posted at: {tweet['createdAt']}")
    
    # Example 3: Get tweet engagement
    print("\n=== Tweet Engagement Example ===")
    tweet_id = results['tweets'][0]['id']  # Use first tweet from search
    
    # Get replies
    replies = client.get_tweet_replies(tweet_id)
    print(f"\nReplies to tweet {tweet_id}:")
    for reply in replies['replies'][:3]:
        print(f"- {reply['text'][:100]}...")
    
    # Get quotes
    quotes = client.get_tweet_quotes(tweet_id)
    print(f"\nQuotes of tweet {tweet_id}:")
    for quote in quotes['tweets'][:3]:
        print(f"- {quote['text'][:100]}...")
    
    # Example 4: Get user mentions
    print(f"\n=== Recent mentions of {username} ===")
    since_time = int((datetime.now() - timedelta(days=7)).timestamp())
    mentions = client.get_user_mentions(username, since_time=since_time)
    for mention in mentions['tweets'][:3]:
        print(f"- {mention['text'][:100]}...")
        print(f"  By: {mention['author']['userName']}")

if __name__ == "__main__":
    main() 